crates/runtime/src/embeddings/index/s3/index.rs

```diff
@@ -62,28 +61,52 @@ pub struct S3Vector {
     pub primary_key: Vec<Field>,
 
     /// Additional columns to add as metadata to the S3 vector index from the original dataset columns.
-    pub metadata_columns: MetadataColumns,
+    /// Note: This does not include the embedding column itself.
+    pub base_metadata_columns: MetadataColumns,
+
+    /// All metadata columns including the embedding column as non-filterable metadata.
+    pub complete_metadata_columns: MetadataColumns,
 
     pub model_name: String,
 
     pub embedding_models: Arc<RwLock<EmbeddingModelStore>>,
 }
 
 impl S3Vector {
+    #[allow(clippy::cast_possible_truncation, clippy::cast_possible_wrap)]
     #[must_use]
     pub fn new(
         table: S3VectorsTable,
         embedded_column: String,
         primary_key: Vec<Field>,
         metadata_columns: MetadataColumns,
         model_name: String,
+        dimension: usize,
         embedding_models: Arc<RwLock<EmbeddingModelStore>>,
     ) -> Self {
+        // Build complete metadata columns including the embedding column as non-filterable
```

Comment:
Why is this necessary? it seems weird to call it a metadata column when its just the main data for the vector index?

Reply:
As it relates to a more generic `SearchIndex`, the embedding vector is metadata (i.e. FTS and other retrieval mechanisms don't have this concept).